interface/wx/webview.h

```diff
@@ -297,6 +450,81 @@ class wxWebViewHandler
         @since 3.1.5
     */
     virtual wxString GetSecurityURL() const;
+
+    /**
+        When using the edge backend handler urls are https urls with a
+        virtual host. As default @c name.wxsite is used as the virtual hostname.
+        If you customize this host use a non existing site (ideally a reserved
+        subdomain of a domain you control). If @c localassests.domain.example is
+        used the handlers content will be available under
+        %https://localassets.domain.example/
+
+        This has to be set @b before registering the handler via
+        wxWebView::RegisterHandler().
+
+        @since 3.3.0
+    */
+    virtual void SetVirtualHost(const wxString& host);
+
+    /**
+        @return The virtual host of this handler
+
+        @see SetVirtualHost()
+        @since 3.3.0
+    */
+    virtual wxString GetVirtualHost() const;
+
+    /**
+        Implementing this method allows for more control over requests from
+        the backend then GetFile(). More details of the request are available
+        from the request object which allows access to URL, method, postdata
+        and headers.
+
+        A response can be send via the response object. The response does not
+        have to be finished from this method and it's possible to be finished
+        asynchronously via wxWebViewHandlerResponse::Finish().
+
+        The following pseudo code demonstrates a typical implementation:
+        @code
+        void StartRequest(const wxWebViewHandlerRequest& request,
+                              wxSharedPtr<wxWebViewHandlerResponse> response) wxOVERRIDE
+        {
+            // Set common headers allowing access from XMLHTTPRequests()
+            response->SetHeader("Access-Control-Allow-Origin", "*");
+            response->SetHeader("Access-Control-Allow-Headers", "*");
+
+            // Handle options request
```

Comment:
Do we _have_ to do this? I.e. do we get `OPTIONS` request even if we don't generate it ourselves? Or is for demonstration purposes only (but then why does the same do it too)?

Reply:
It depends on the context, but a `XMLHTTPRequest` generates (regardless of the requests HTTP method) a `OPTIONS` request from the browser to check cross site security.